ext/standard/php_array.h

```diff
@@ -34,6 +34,19 @@ PHPAPI zend_long php_count_recursive(HashTable *ht);
 PHPAPI bool php_array_data_shuffle(const php_random_algo *algo, php_random_status *status, zval *array);
 PHPAPI bool php_array_pick_keys(const php_random_algo *algo, php_random_status *status, zval *input, zend_long num_req, zval *retval, bool silent);
 
+#define EXTR_OVERWRITE			0
+#define EXTR_SKIP				1
+#define EXTR_PREFIX_SAME		2
+#define	EXTR_PREFIX_ALL			3
+#define	EXTR_PREFIX_INVALID		4
+#define	EXTR_PREFIX_IF_EXISTS	5
+#define	EXTR_IF_EXISTS			6
+
+#define EXTR_REFS				0x100
+
+#define CASE_LOWER				0
+#define CASE_UPPER				1
```

Comment:
php_array.h is a public header; I'm afraid exposing these macros may be an unnecessary BC break. Can we add them to an internal header instead? Otherwise, we should probably prefix these macros with `PHP_`.

Reply:
I think prefixing with PHP_ is the most reasonable - I don't see why they should _not_ be in a public header, but I agree about the conflict potential of a name like "CASE_LOWER".

Reply:
> Can we add them to an internal header instead?

I prefer what Bob said, too :) But if you still think that adding a private header is worthwhile, then I'm ok to do so.